### PR TITLE
Imx8mm

### PR DESCRIPTION
**Pull request by js731ca**
two fixes for imx8mm

### CODE REVIEW

File: arch/arm/mach-imx/imx8m.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ static int imx8m_init(const char *cputypestr)
 	pr_info("%s unique ID: %llx\n", cputypestr, imx8m_uid());
 
 	if (IS_ENABLED(CONFIG_ARM_SMCCC) &&
-	    IS_ENABLED(CONFIG_FIRMWARE_IMX8MQ_ATF)) {
+	    IS_ENABLED(CONFIG_FIRMWARE_IMX8MM_ATF)) {
```
Comment (a3f) [inline arch/arm/mach-imx/imx8m.c:64]: As the function is common to all of i.MX8M, both IMX8MM and IMX8MQ defines are out of place. Either add a new CONFIG_FIRMWARE_IMX8M_ATF symbol that's select by both or replace with a `get_el() == 2` check.

File: arch/arm/boards/nxp-imx8mm-evk/lowlevel.c [hunk 2]
```diff
@@ -61,9 +66,9 @@ static void power_init_board(void)
 	imx8mm_early_clock_init();
 	imx8m_ccgr_clock_enable(IMX8M_CCM_CCGR_I2C1);
 
-	i2c = imx8m_i2c_early_init(IOMEM(MX8MQ_I2C1_BASE_ADDR));
+	i2c = imx8m_i2c_early_init(IOMEM(MX8MM_I2C1_BASE_ADDR));
 
-	pmic_configure(i2c, 0x4b, bd71837_cfg, ARRAY_SIZE(bd71837_cfg));
+	pmic_configure(i2c, 0x25, pca9450_cfg, ARRAY_SIZE(pca9450_cfg));
```
Comment (a3f) [inline arch/arm/boards/nxp-imx8mm-evk/lowlevel.c:71]: Is your board the LPDDR4 or DDR4 variant? Anyhow, please keep old support in place. Check out i.MX8MN-EVK support for an example of how to detect whether to use Rohm or PCA PMIC at runtime.

File: arch/arm/boards/nxp-imx8mp-evk/lowlevel.c [hunk 3]
```diff
@@ -72,7 +72,7 @@ static void power_init_board(void)
 	imx8mp_setup_pad(MX8MP_PAD_I2C1_SCL__I2C1_SCL | I2C_PAD_CTRL);
 	imx8mp_setup_pad(MX8MP_PAD_I2C1_SDA__I2C1_SDA | I2C_PAD_CTRL);
 
-	imx8mm_early_clock_init();
+	imx8mp_early_clock_init();
```
Comment (a3f) [inline arch/arm/boards/nxp-imx8mp-evk/lowlevel.c:75]: Please split up into three patches: 8MM, 8MP,  8M -generic code.

File: dts/src/arm64/freescale/imx8mm-evk.dts [hunk 4]
```diff
@@ -17,6 +17,23 @@
 	};
 };
 
+
```
Comment (a3f) [inline dts/src/arm64/freescale/imx8mm-evk.dts:20]: The whole `dts/` directory is imported wholesale from Linux and synced after every `-rc` and final release. Patches to these files should be submitted upstream to the SoC's Linux device tree maintainers.

When submitting it there, it should be a new file and not affect the old DT and it should have its own compatible. As an interim solution, you can add the DT to `arch/arm/dts` which contains the barebox DTs.

For more info, see: https://www.barebox.org/doc/latest/devicetree/index.html#barebox-device-trees

**Review by a3f — CHANGES_REQUESTED** [review]
Usual contribution model is via mailing list. If you can use git-send-email, please submit your patches to the mailing list instead. This ensures more people will see your patches and comment on what might need changing.

Also, please adopt normal convention for commit messages, e.g.:

```
ARM: i.MX8M: imx8mm-evk: add support for new PCA9650 LPDDR4 variant

Existing EVK support is for LPDDR4 variant wit Rohm BD71837 PMIC,
but newer EVKs are fitted with PCA9450 on SoM. There's no upstream DT yet
.... etc.


S-o-b:
```

File: arch/arm/mach-imx/imx8m.c (hunk 1 above)
Comment (js731ca) [reply arch/arm/mach-imx/imx8m.c:64]: looking at firmware/Makefile - wich adds a per variant bl31.bin - i think it should be an || of all imx8m variants instead
'grep -rniI get_el' didn't turn up any matches - what did you have in mind there?

**Comment by js731ca** [comment]
> Usual contribution model is via mailing list.

that's an "aaaaw, and there i was happy to find 'the' official github repository" with all it's niceties (webinterface, PR handling, ...)
should have read the readme more thoroughly - so i'll close this PR and push v2 to the ML?

File: dts/src/arm64/freescale/imx8mm-evk.dts (hunk 4 above)
Comment (js731ca) [reply dts/src/arm64/freescale/imx8mm-evk.dts:20]: as you probably noticed - patches are pushed (or tried to) to the linux kernel ML
i'll drop the added dts/patch here; wait for the kernel/dts sync

File: arch/arm/mach-imx/imx8m.c (hunk 1 above)
Comment (a3f) [reply arch/arm/mach-imx/imx8m.c:64]: > looking at firmware/Makefile - wich adds a per variant bl31.bin - i think it should be an || of all imx8m variants instead

That's equivalent to having all SoC-specific symbols select one symbol and check against that in code.

> 'grep -rniI get_el' didn't turn up any matches - what did you have in mind there?

Sorry, it's called `current_el()`

File: dts/src/arm64/freescale/imx8mm-evk.dts (hunk 4 above)
Comment (a3f) [reply dts/src/arm64/freescale/imx8mm-evk.dts:20]: You can for now add the evkb DT to `arch/arm/dts` in barebox and turn it into an `#include <arm64/freescale/imx8mm-evkb.dts>` once the changes make it into a kernel release that's imported into barebox.

**Comment by a3f** [comment]
>  should have read the readme more thoroughly - so i'll close this PR and push v2 to the ML?

Sascha occasionally merged patches via Github for contributors that have problems sending unmangled mails, but it's not the norm. This is just a mirror. Let's continue on the mailing list. :)

**Comment by a3f** [comment]
FTR, this is now resolved by https://github.com/saschahauer/barebox/commit/49b0c622eff312b52c26835d6361f025d16be16b, so this PR can be closed.

**Comment by js731ca** [comment]
Thanks! (-:

**Pull request closed without merging**